blackbird_python/blackbird/program.py

```diff
@@ -314,25 +313,47 @@ def serialize(self):
 
         script = ["name {}".format(self.name), "version {}".format(self.version)]
 
-        if self.target["name"] is not None:
-            array_insert += 1
-            options = ""
+        # add target and type to the script
+        for n, md in [("target", self.target), ("type", self.programtype)]:
+            if md["name"] is not None:
+                array_insert += 1
+                options = ""
 
-            if self.target["options"]:
-                # if the target has options, compile them into
-                # the expected syntax
-                option_strings = [
-                    "{}={}".format(k, v) if not isinstance(v, str) else '{}="{}"'.format(k, v)
-                    for k, v in self.target["options"].items()
-                ]
-                options = " ({})".format(", ".join(option_strings))
+                if md["options"]:
+                    # if the target has options, compile them into
+                    # the expected syntax
+                    option_strings = [
+                        "{}={}".format(k, v) if not isinstance(v, str) else '{}="{}"'.format(k, v)
+                        for k, v in md["options"].items()
+                    ]
+                    options = " ({})".format(", ".join(option_strings))
 
-            # add target metadata
-            script.append("target {}{}".format(self.target["name"], options))
+                # add target metadata
+                script.append("{} {}{}".format(n, md["name"], options))
 
         # line break
         script.append("")
 
+        # add variables to the script (works for all type of programs, but some
+        # tests would need to be updated; if wanted, simply remove the next line)
```

Comment:
Probably a good idea to remove the comment regarding tests before merging this (it feels odd for the source code to reference the tests)